ui/components/src/Tooltip/Tooltip.tsx

```diff
@@ -16,28 +16,21 @@ import { Box, Portal } from '@mui/material';
 import { ECharts as EChartsInstance } from 'echarts/core';
 import { EChartsDataFormat } from '../model/graph';
 import { getFocusedSeriesData } from './focused-series';
-import {
-  CursorCoordinates,
-  CursorData,
-  TooltipData,
-  TOOLTIP_MAX_HEIGHT,
-  TOOLTIP_MAX_WIDTH,
-  useMousePosition,
-} from './tooltip-model';
+import { CursorCoordinates, TOOLTIP_MAX_HEIGHT, TOOLTIP_MAX_WIDTH, useMousePosition } from './tooltip-model';
 import { TooltipContent } from './TooltipContent';
+import { assembleTransform } from './utils';
 
 interface TooltipProps {
   chartRef: React.MutableRefObject<EChartsInstance | undefined>;
-  tooltipData: TooltipData;
   chartData: EChartsDataFormat;
   wrapLabels?: boolean;
 }
 
-export function Tooltip(props: TooltipProps) {
-  const { chartRef, chartData } = props;
+const Tooltip = React.memo(function Tooltip({ chartRef, chartData, wrapLabels }: TooltipProps) {
   const [pinnedPos, setPinnedPos] = useState<CursorCoordinates | null>(null);
   const mousePos = useMousePosition();
-  if (mousePos === null) return null;
+  if (mousePos === null || mousePos.target === null) return null;
+  if ((mousePos.target as HTMLElement).tagName !== 'CANVAS') return null;
```

Comment:
main improvement is here to first check if the cursor is above a canvas element before calling getFocusedSeriesData